src/vs/platform/terminal/test/common/capabilities/commandDetection/promptInputModel.test.ts

```diff
@@ -698,6 +699,66 @@ suite('PromptInputModel', () => {
 			await writePromise('"a"');
 			// HACK: Trailing whitespace is due to flaky detection in wrapped lines (but it doesn't matter much)
 			await assertPromptInput(`echo "a"| `);
+			await writePromise('\n\r\ b');
+			await assertPromptInput(`echo "a"\n b|`);
+			await writePromise('\n\r\ c');
+			await assertPromptInput(`echo "a"\n b\n c|`);
+		});
+	});
+	suite('multi-line wrapped (continuation prompt)', () => {
+		test('basic wrapped line', async () => {
+			xterm.resize(5, 10);
+			promptInputModel.setContinuationPrompt('∙ ');
+			await writePromise('$ ');
+			fireCommandStart();
+			await assertPromptInput('|');
+
+			await writePromise('ech');
+			await assertPromptInput(`ech|`);
+
+			await writePromise('o ');
+			await assertPromptInput(`echo |`);
+
+			await writePromise('"a"');
+			// HACK: Trailing whitespace is due to flaky detection in wrapped lines (but it doesn't matter much)
+			await assertPromptInput(`echo "a"| `);
+			await writePromise('\n\r\∙ ');
+			await assertPromptInput(`echo "a"\n|`);
+			await writePromise('b');
+			await assertPromptInput(`echo "a"\nb|`);
+			await writePromise('\n\r\∙ ');
+			await assertPromptInput(`echo "a"\nb\n|`);
+			await writePromise('c');
+			await assertPromptInput(`echo "a"\nb\nc|`);
+			await writePromise('\n\r\∙ ');
+			await assertPromptInput(`echo "a"\nb\nc\n|`);
+		});
+	});
+	suite('multi-line wrapped fish', () => {
```

Comment:
Don't think we cover this case?

```
[I] meganrogge@Megans-MacBook-Pro ~ (main|BISECTING)> echo "hello
                                                      world"|
```

Reply:
think that's what this does? https://github.com/microsoft/vscode/blob/4ba1f5fec254ebb5b1c32cbdd24ea0f42eca54f3/src/vs/platform/terminal/test/common/capabilities/commandDetection/promptInputModel.test.ts#L750-L762